### ISSUE TITLE
[BUG] - Import error with new version of RAPIDS

### ISSUE THREAD

**Issue by johnhickey22** [scverse/rapids_singlecell#96]
**Description**
Since the RAPIDS team said that they fixed the number of clusters from leiden clustering ([issue](https://github.com/rapidsai/cugraph/issues/3749#issuecomment-1802458723)), I tried to install this nightly version. But when I go to use rapids_singlecell, I get this error:

```
---> 10 import rapids_singlecell as rsc

14 frames
[/usr/local/lib/python3.10/dist-packages/cffi/api.py](https://localhost:8080/#) in __init__(self, backend)
     52                 if hasattr(backend, '__file__'):
     53                     # CPython
---> 54                     raise Exception("Version mismatch: this is the 'cffi' package version %s, located in %r.  When we import the top-level '_cffi_backend' extension module, we get version %s, located in %r.  The two versions should be equal; check your installation." % (
     55                         __version__, __file__,
     56                         backend.__version__, backend.__file__))

Exception: Version mismatch: this is the 'cffi' package version 1.16.0, located in '/usr/local/lib/python3.10/dist-packages/cffi/api.py'.  When we import the top-level '_cffi_backend' extension module, we get version 1.15.0, located in '/usr/local/lib/python3.10/site-packages/_cffi_backend.cpython-310-x86_64-linux-gnu.so'.  The two versions should be equal; check your installation.
```

Any thoughts on how to get around this?

**Steps/Code to reproduce bug**
I am working in Google Colab to test things out and currently have the following packages installed:

Package                       Version
----------------------------- ------------
aiohttp                       3.8.6
aiosignal                     1.3.1
anndata                       0.10.3
annotated-types               0.6.0
anyio                         3.7.1
appdirs                       1.4.4
argon2-cffi                   23.1.0
argon2-cffi-bindings          21.2.0
array-api-compat              1.4
arrow                         1.3.0
async-timeout                 4.0.3
attrs                         23.1.0
backports.functools-lru-cache 1.6.5
beautifulsoup4                4.12.2
bleach                        6.1.0
blinker                       1.7.0
bokeh                         3.3.1
branca                        0.7.0
brotlipy                      0.7.0
cached-property               1.5.2
cachetools                    5.3.2
certifi                       2023.7.22
cffi                          1.15.0
charset-normalizer            3.1.0
click                         8.1.7
click-plugins                 1.1.1
cligj                         0.7.2
cloudpickle                   3.0.0
colorama                      0.4.6
colorcet                      3.0.1
conda                         23.1.0
conda-package-handling        2.0.2
conda_package_streaming       0.7.0
confluent-kafka               1.9.2
contourpy                     1.2.0
cubinlinker                   0.3.0
cucim                         23.12.0a18
cuda-python                   11.8.3
cudf                          23.12.0a704
cudf-kafka                    23.12.0a704
cugraph                       23.12.0a92
cuml                          23.12.0a44
cuproj                        23.12.0a22
cupy                          12.2.0
cuspatial                     23.12.0a22
custreamz                     23.12.0a704
cuxfilter                     23.12.0a10
cycler                        0.12.1
cytoolz                       0.12.2
dask                          2023.9.3
dask-cuda                     23.12.0a29
dask-cudf                     23.12.0a704
dask_sql                      2023.10.1
datashader                    0.16.0
decorator                     5.1.1
decoupler                     1.5.0
defusedxml                    0.7.1
distributed                   2023.9.3
entrypoints                   0.4
exceptiongroup                1.1.3
fastapi                       0.104.1
fastjsonschema                2.19.0
fastrlock                     0.8.2
Fiona                         1.9.3
folium                        0.15.0
fonttools                     4.44.3
fqdn                          1.5.1
frozenlist                    1.4.0
fsspec                        2023.10.0
gcsfs                         2023.10.0
GDAL                          3.6.4
geopandas                     0.14.1
google-api-core               2.14.0
google-auth                   2.23.4
google-auth-oauthlib          1.1.0
google-cloud-core             2.3.3
google-cloud-storage          2.13.0
google-crc32c                 1.1.2
google-resumable-media        2.6.0
googleapis-common-protos      1.61.0
grpcio                        1.59.2
h11                           0.14.0
h2                            4.1.0
h5py                          3.10.0
harmonypy                     0.0.9
holoviews                     1.18.1
hpack                         4.0.0
httpcore                      1.0.2
httpx                         0.25.1
hyperframe                    6.0.1
idna                          3.4
igraph                        0.10.8
imagecodecs                   2023.9.18
imageio                       2.31.5
importlib-metadata            6.8.0
importlib-resources           6.1.1
isoduration                   20.11.0
Jinja2                        3.1.2
joblib                        1.3.2
jsonpointer                   2.4
jsonschema                    4.20.0
jsonschema-specifications     2023.11.1
jupyter_client                8.6.0
jupyter_core                  5.5.0
jupyter-events                0.9.0
jupyter_server                2.10.1
jupyter_server_proxy          4.1.0
jupyter_server_terminals      0.4.4
jupyterlab-pygments           0.2.2
kiwisolver                    1.4.5
lazy_loader                   0.3
leidenalg                     0.10.1
libmambapy                    1.5.0
linkify-it-py                 2.0.0
llvmlite                      0.40.1
locket                        1.0.0
lz4                           4.3.2
mamba                         1.5.0
mapclassify                   2.6.1
Markdown                      3.5.1
markdown-it-py                3.0.0
MarkupSafe                    2.1.3
matplotlib                    3.8.0
mdit-py-plugins               0.4.0
mdurl                         0.1.0
mistune                       3.0.2
msgpack                       1.0.6
multidict                     6.0.4
multipledispatch              0.6.0
munch                         4.0.0
munkres                       1.1.4
natsort                       8.4.0
nbclient                      0.8.0
nbconvert                     7.11.0
nbformat                      5.9.2
networkx                      3.2.1
numba                         0.57.1
numpy                         1.24.4
nvtx                          0.2.8
oauthlib                      3.2.2
overrides                     7.4.0
packaging                     23.2
pandas                        1.5.3
pandocfilters                 1.5.0
panel                         1.3.1
param                         2.0.1
partd                         1.4.1
patsy                         0.5.3
Pillow                        10.0.1
pip                           23.0.1
pkgutil_resolve_name          1.3.10
platformdirs                  4.0.0
pluggy                        1.0.0
prometheus-client             0.18.0
prompt-toolkit                3.0.41
protobuf                      4.24.4
psutil                        5.9.5
ptxcompiler                   0.8.1
ptyprocess                    0.7.0
pyarrow                       14.0.1
pyasn1                        0.5.0
pyasn1-modules                0.3.0
pycosat                       0.6.4
pycparser                     2.21
pyct                          0.4.6
pydantic                      2.5.1
pydantic_core                 2.14.3
pyee                          8.1.0
Pygments                      2.16.1
PyJWT                         2.8.0
pylibcugraph                  23.12.0a92
pylibraft                     23.12.0a85
pynndescent                   0.5.10
pynvml                        11.4.1
pyOpenSSL                     23.1.1
pyparsing                     3.1.1
pyppeteer                     1.0.2
pyproj                        3.6.1
PySocks                       1.7.1
python-dateutil               2.8.2
python-json-logger            2.0.7
pytz                          2023.3.post1
pyu2f                         0.1.5
pyviz_comms                   3.0.0
PyWavelets                    1.4.1
PyYAML                        6.0.1
pyzmq                         25.1.1
raft-dask                     23.12.0a85
rapids_singlecell             0.9.3
referencing                   0.31.0
requests                      2.28.2
requests-oauthlib             1.3.1
rfc3339-validator             0.1.4
rfc3986-validator             0.1.1
rich                          13.7.0
rmm                           23.12.0a22
rpds-py                       0.13.0
rsa                           4.9
Rtree                         1.1.0
ruamel.yaml                   0.17.21
ruamel.yaml.clib              0.2.7
scanpy                        1.9.6
scikit-image                  0.21.0
scikit-learn                  1.3.2
scikit-misc                   0.3.0
scipy                         1.11.3
seaborn                       0.12.2
Send2Trash                    1.8.2
session-info                  1.0.0
setuptools                    65.6.3
shapely                       2.0.2
simpervisor                   1.0.0
six                           1.16.0
sniffio                       1.3.0
sortedcontainers              2.4.0
soupsieve                     2.5
starlette                     0.27.0
statsmodels                   0.14.0
stdlib-list                   0.9.0
streamz                       0.6.4
tabulate                      0.9.0
tbb                           2021.11.0
tblib                         2.0.0
terminado                     0.18.0
texttable                     1.7.0
threadpoolctl                 3.2.0
tifffile                      2023.9.26
tinycss2                      1.2.1
toolz                         0.12.0
tornado                       6.3.3
tqdm                          4.65.0
traitlets                     5.13.0
treelite                      3.9.1
treelite-runtime              3.9.1
types-python-dateutil         2.8.19.14
typing_extensions             4.8.0
typing-utils                  0.1.0
tzlocal                       5.2
uc-micro-py                   1.0.1
ucx-py                        0.35.0a13
umap-learn                    0.5.4
unicodedata2                  15.1.0
uri-template                  1.3.0
urllib3                       1.26.15
uvicorn                       0.24.0
wcwidth                       0.2.10
webcolors                     1.13
webencodings                  0.5.1
websocket-client              1.6.4
websockets                    10.4
wheel                         0.40.0
xarray                        2023.10.1
xgboost                       1.7.6
xyzservices                   2023.10.1
yarl                          1.9.2
zict                          3.0.0
zipp                          3.17.0
zstandard                     0.19.0

Thanks!

**Comment by Intron7** [comment]
The PR for the leiden-fix is not yet merged, at least the one they link in the issue. I have no experience working with google colab. But since this is still alpha software, things will probably be fixed by the main release. I also don't think that this a rapids-singlecell issue but a rapids-alpha with colab issue. 
If the issue persists with the stable RAPIDS-23.12 and you are able to use this rapids version on colab without installing rsc, I'll continue to look into it but for now I can't support alpha-software.

**Comment by Intron7** [comment]
I tested the new version of rapids-23.12 nightly with rapids-singlecell after the leiden clustering got fixed. The bug might only be on colab because for me it works.

**Comment by johnhickey22** [comment]
Based on your messages, I just tried pip upgrade 'cffi' and this worked now when they merged the fix into the nightly branch. So seems it was a Colab issue, but still working nicely for me now too. Thanks!
